File: avalon-support/avalon-oauth2-support/avalon-oauth2-common/src/test/java/io/github/lishangbu/avalon/oauth2/common/userdetails/UserInfoTest.java
package io.github.lishangbu.avalon.oauth2.common.userdetails;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.List;
import org.junit.jupiter.api.Test;
import org.springframework.security.core.authority.SimpleGrantedAuthority;

class UserInfoTest {

    @Test
    void constructorCopiesStateAndAttributesAreMutable() {
        UserInfo user = new UserInfo(
                "alice", "pwd", List.of(new SimpleGrantedAuthority("ROLE_USER")));

        user.getAdditionalParameters().put("key", "value");

        assertEquals("alice", user.getUsername());
        assertEquals("alice", user.getName());
        assertEquals("value", user.getAttributes().get("key"));
        assertTrue(user.isAccountNonExpired());
        assertTrue(user.isAccountNonLocked());
        assertTrue(user.isCredentialsNonExpired());
        assertTrue(user.isEnabled());
    }

    @Test
    void fullConstructorSetsFlags() {
        UserInfo user = new UserInfo(
                "bob",
                null,
                true,
                false,
                true,
                false,
                List.of(new SimpleGrantedAuthority("ROLE_ADMIN")));

        assertEquals("bob", user.getUsername());
        assertTrue(user.isEnabled());
        assertFalse(user.isAccountNonExpired());
        assertTrue(user.isCredentialsNonExpired());
        assertFalse(user.isAccountNonLocked());
    }

    @Test
    void toStringContainsRelevantFields() {
        UserInfo user = new UserInfo(
                "charlie", "pwd", List.of(new SimpleGrantedAuthority("ROLE_USER")));
        user.getAdditionalParameters().put("foo", "bar");

        String text = user.toString();

        assertTrue(text.contains("Username=charlie"));
        assertTrue(text.contains("AdditionalParameters"));
        assertTrue(text.contains("Granted Authorities"));
    }
}

File: avalon-platform/avalon-security/avalon-oauth2-common/src/test/kotlin/io/github/lishangbu/avalon/oauth2/common/userdetails/UserInfoTest.kt
package io.github.lishangbu.avalon.oauth2.common.userdetails

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import org.springframework.security.core.authority.SimpleGrantedAuthority

class UserInfoTest {
    @Test
    fun constructorCopiesStateAndAttributesAreMutable() {
        val user = UserInfo("alice", "pwd", listOf(SimpleGrantedAuthority("ROLE_USER")))

        user.additionalParameters["key"] = "value"

        assertEquals("alice", user.username)
        assertEquals("alice", user.name)
        assertEquals("value", user.attributes["key"])
        assertTrue(user.isAccountNonExpired)
        assertTrue(user.isAccountNonLocked)
        assertTrue(user.isCredentialsNonExpired)
        assertTrue(user.isEnabled)
    }

    @Test
    fun fullConstructorSetsFlags() {
        val user =
            UserInfo(
                "bob",
                null,
                true,
                false,
                true,
                false,
                listOf(SimpleGrantedAuthority("ROLE_ADMIN")),
            )

        assertEquals("bob", user.username)
        assertTrue(user.isEnabled)
        assertFalse(user.isAccountNonExpired)
        assertTrue(user.isCredentialsNonExpired)
        assertFalse(user.isAccountNonLocked)
    }

    @Test
    fun toStringContainsRelevantFields() {
        val user = UserInfo("charlie", "pwd", listOf(SimpleGrantedAuthority("ROLE_USER")))
        user.additionalParameters["foo"] = "bar"

        val text = user.toString()

        assertTrue(text.contains("Username=charlie"))
        assertTrue(text.contains("AdditionalParameters"))
        assertTrue(text.contains("Granted Authorities"))
    }
}
